dsc_lib/src/dscresources/command_resource.rs

```diff
@@ -583,62 +573,110 @@ pub fn invoke_command(executable: &str, args: Option<Vec<String>>, input: Option
+        Err(e) => {
+            return Err(DscError::CommandOperation(e.to_string(), executable.to_string()))
+        }
+    };
+
+    let stdout = child.stdout.take().expect("child did not have a handle to stdout");
+    let stderr = child.stderr.take().expect("child did not have a handle to stderr");
+    let mut stdout_reader = BufReader::new(stdout).lines();
+    let mut stderr_reader = BufReader::new(stderr).lines();
+
     if let Some(input) = input {
         trace!("Writing to command STDIN: {input}");
-        // pipe to child stdin in a scope so that it is dropped before we wait
-        // otherwise the pipe isn't closed and the child process waits forever
-        let Some(mut child_stdin) = child.stdin.take() else {
-            return Err(DscError::CommandOperation("Failed to open stdin".to_string(), executable.to_string()));
-        };
-        child_stdin.write_all(input.as_bytes())?;
-        child_stdin.flush()?;
+        let mut stdin = child.stdin.take().expect("child did not have a handle to stdin");
+        stdin.write_all(input.as_bytes()).await.expect("could not write to stdin");
+        drop(stdin);
     }
 
-    let Some(mut child_stdout) = child.stdout.take() else {
-        return Err(DscError::CommandOperation("Failed to open stdout".to_string(), executable.to_string()));
+    let child_id: u32 = match child.id() {
+        Some(id) => id,
+        None => {
+            return Err(DscError::CommandOperation("Can't get child process id".to_string(), executable.to_string()));
+        }
     };
-    let mut stdout_buf = Vec::new();
-    child_stdout.read_to_end(&mut stdout_buf)?;
 
-    let Some(mut child_stderr) = child.stderr.take() else {
-        return Err(DscError::CommandOperation("Failed to open stderr".to_string(), executable.to_string()));
-    };
-    let mut stderr_buf = Vec::new();
-    child_stderr.read_to_end(&mut stderr_buf)?;
-
-    let exit_status = child.wait()?;
-    let exit_code = exit_status.code().unwrap_or(EXIT_PROCESS_TERMINATED);
-    let stdout = String::from_utf8_lossy(&stdout_buf).to_string();
-    let stderr = String::from_utf8_lossy(&stderr_buf).to_string();
-    if !stdout.is_empty() {
-        trace!("STDOUT returned: {}", &stdout);
-    }
-    let cleaned_stderr = if stderr.is_empty() {
-        stderr
-    } else {
-        trace!("STDERR returned data to be traced");
-        log_resource_traces(executable, &child.id(), &stderr);
-        // TODO: remove logged traces from STDERR
-        String::new()
-    };
+    let child_task = tokio::spawn(async move {
+        child.wait().await
+    });
 
-    if exit_code != 0 {
-        if let Some(exit_codes) = exit_codes {
-            if let Some(error_message) = exit_codes.get(&exit_code) {
-                return Err(DscError::CommandExitFromManifest(executable.to_string(), exit_code, error_message.to_string()));
+    let stdout_task = tokio::spawn(async move {
+        let mut stdout_result = String::with_capacity(1024*1024);
```

Comment:
Since 1024*1024 is used more than once, you should define it as a const and include a comment why you chose that value

Reply:
Updated.